File: go/example/submit_url_interactive.go
// Copyright (C) 2020-2021 Hatching B.V.
// All rights reserved.

package main

import (
	"context"
	"fmt"

	triage "github.com/hatching/triage/go"
)

const (
	Triage = "https://api.tria.ge"
	Token  = "<YOUR-APIKEY-HERE>"
)

func main() {
	client := triage.NewClientWithRootURL(Token, Triage)

	sample, err := client.SubmitSampleURL(
		context.Background(), "http://www.google.nl", true, nil,
	)
	if err != nil {
		panic(err)
	}
	fmt.Println("sample:", sample.ID)

	err = client.SetSampleProfileAutomatically(
		context.Background(), sample.ID, []string{},
	)
	if err != nil {
		panic(err)
	}
	fmt.Println("Profile set for", sample.ID)
}

File: python/example/submit_url_interactive.py
# Copyright (C) 2020 Hatching B.V
# All rights reserved.

import io
import time

from triage import Client

url = "https://api.tria.ge"
token = "<YOUR-APIKEY-HERE>"

c = Client(token, root_url=url)
r = c.submit_sample_url("http://google.com", interactive=True)
print(r)
print("waiting..") # triage takes some time to process
profile = input("profile: ")
time.sleep(1)
r = c.set_sample_profile(r["id"], [{"profile": profile}])
print(r)
print("submitted")
